### PR TITLE
Upg: improve builder UX when using a template for instructions

### PR DESCRIPTION
**Pull request by Fraggle**
## Description

Animate as a typewriter the instructions for an assistant when using a template, then focus on it and put cursor at the end.

In respect to [this comment](https://github.com/dust-tt/dust/pull/6722#discussion_r1711185660) we investigated a bit more and found a couple things:
- `editor` from `useEditor` will mutate (eg: as part of hook dependencies) on content change ([doc](https://tiptap.dev/docs/guides/performance))
- even when using an extension with a lower access to the input component, it still mutate.
- hence, useMemo() (such as for `xxxEditorService`) will probably not do what we expect.
- other solutions to achieve the typewriter effect will not work for multi-lines content (we cannot access individual letters)

We decided to keep the implementation like that for now and revisit later.

https://github.com/user-attachments/assets/1c676de1-271d-49ff-bbbd-f48a43e42106

## Risk

A bug could led the input box to be uneditable.

## Deploy Plan

Deploy `front`

### CODE REVIEW

**Review by fontanierh — APPROVED** [review]
LGTM

File: front/components/assistant_builder/InstructionScreen.tsx [hunk 1]
```diff
@@ -126,19 +130,63 @@ export function InstructionScreen({
...
+          instructions: plainText,
+        }));
+      }
     },
   });
   const editorService = useInstructionEditorService(editor);
 
+  const [letterIndex, setLetterIndex] = useState(0);
+
+  useEffect(() => {
```
Comment (flvndvd) [inline front/components/assistant_builder/InstructionScreen.tsx:153]: Looking more at this, it's pretty heavy re-rendering. For each letter we will re-render this component. Either you move this outside of this component so the useEffect is not part of a rendered component but only uses the TipTap API to play the animation. This use case is also often achieve using `requestanimationframe`. Also I'm wondering if we could not leverage CSS styles here, with something like this:

```
    editor.chain()
      .setContent(`<span class="typewriter-effect">${textContent}</span>`)
      .focus("end")
      .run();

    // Set editable to false while typewriter effect is running
    editor.setEditable(false);

    // Once the animation is complete, set editable to true
    const animationDuration = 4000; // Duration should match the CSS animation duration
    const timeoutId = setTimeout(() => {
      editor.setEditable(true);
      setDoTypewriterEffect(false);
    }, animationDuration);
```

**Review by flvndvd — COMMENTED** [review]
The gif is 🔥, left a global comment as it might create too many re-renderings.

File: front/components/assistant_builder/InstructionScreen.tsx (hunk 1 above)
Comment (Fraggle) [reply front/components/assistant_builder/InstructionScreen.tsx:155]: Added some comments and we transcript our findings in PR description.

**Pull request merged**